app/models/policy_version.rb

```diff
@@ -86,6 +86,58 @@ def before_save
     self.policy_sha256 = Digest::SHA256.hexdigest(policy_text)
   end
 
+  def after_save
+    log_versions_off_limit
+    delete_versions_off_limit
+  rescue => e
+    Rails.logger.error("Error while ensuring policy '#{id}' versions retention limit: #{e.inspect}")
+  end
+
+  def log_versions_off_limit
+    versions_off_limit.all.each do |policy_version|
+      Rails.logger.debug(
+        "Deleting policy version: #{policy_version.slice(
```

Comment:
@shulifink

Reply:
Isn't `versions_off_limit` the current set of policy versions we want to keep?

Reply:
no its the ones we delete

Reply:
@eranha - can you show an example of the error with all the variables filled in?

Reply:
@shulifink 
` Error while ensuring policy 'cucumber:policy:root' versions retention limit: #<Sequel::DatabaseError: PG::SyntaxError: ERROR:  syntax error at or near "WITH_"`

Reply:
Error while enforcing version retention limit for policy 'cucumber:policy:root'.